plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -2,100 +2,55 @@
 package azure_data_explorer
 
 import (
-	"bytes"
-	"context"
 	_ "embed"
-	"errors"
 	"fmt"
-	"strings"
 	"time"
 
-	"github.com/Azure/azure-kusto-go/kusto"
-	kustoerrors "github.com/Azure/azure-kusto-go/kusto/data/errors"
 	"github.com/Azure/azure-kusto-go/kusto/ingest"
-	"github.com/Azure/azure-kusto-go/kusto/kql"
+	"github.com/influxdata/telegraf/config"
+	adx_common "github.com/influxdata/telegraf/plugins/common/adx"
+	"github.com/influxdata/telegraf/plugins/serializers/json"
 
 	"github.com/influxdata/telegraf"
-	"github.com/influxdata/telegraf/config"
-	"github.com/influxdata/telegraf/internal"
-	"github.com/influxdata/telegraf/internal/choice"
 	"github.com/influxdata/telegraf/plugins/outputs"
-	"github.com/influxdata/telegraf/plugins/serializers/json"
 )
 
 //go:embed sample.conf
 var sampleConfig string
 
 type AzureDataExplorer struct {
-	Endpoint        string          `toml:"endpoint_url"`
-	Database        string          `toml:"database"`
-	Log             telegraf.Logger `toml:"-"`
-	Timeout         config.Duration `toml:"timeout"`
-	MetricsGrouping string          `toml:"metrics_grouping_type"`
-	TableName       string          `toml:"table_name"`
-	CreateTables    bool            `toml:"create_tables"`
-	IngestionType   string          `toml:"ingestion_type"`
-	serializer      telegraf.Serializer
-	kustoClient     *kusto.Client
-	metricIngestors map[string]ingest.Ingestor
+	adx_common.Config
+	Log        telegraf.Logger `toml:"-"`
+	serializer telegraf.Serializer
+	Client     *adx_common.Client
 }
 
-const (
-	tablePerMetric = "tablepermetric"
-	singleTable    = "singletable"
-	// These control the amount of memory we use when ingesting blobs
-	bufferSize = 1 << 20 // 1 MiB
-	maxBuffers = 5
-)
-
-const managedIngestion = "managed"
-const queuedIngestion = "queued"
-
 func (*AzureDataExplorer) SampleConfig() string {
 	return sampleConfig
 }
 
+func (adx *AzureDataExplorer) SetSerializer(serializer telegraf.Serializer) {
+	adx.serializer = serializer
+}
+
 // Initialize the client and the ingestor
 func (adx *AzureDataExplorer) Connect() error {
-	conn := kusto.NewConnectionStringBuilder(adx.Endpoint).WithDefaultAzureCredential()
-	// Since init is called before connect, we can set the connector details here including the type. This will be used for telemetry and tracing.
-	conn.SetConnectorDetails("Telegraf", internal.ProductToken(), "", "", false, "")
-	client, err := kusto.New(conn)
-	if err != nil {
-		return err
+	var err error
+	if adx.Client, err = adx.Config.NewClient("Kusto.Telegraf", adx.Log); err != nil {
+		return fmt.Errorf("error creating new client. Error: %w", err)
```

Comment:
Please use the following style
```suggestion
		return fmt.Errorf("creating new client failed: %w", err)
```
as this avoids punctuation etc. and makes it easy to follow the path...